File: java/leetcode_grind/Day589.java
package leetcode_grind;

public class Day589 {
    // https://leetcode.com/problems/find-center-of-star-graph/description/?envType=daily-question&envId=2024-06-27
    static class Solution1 {
        public int findCenter(int[][] edges) {
            var e1 = edges[0];
            var e2 = edges[1];
            if (e1[0] == e2[0] || e1[0] == e2[1]) {
                return e1[0];
            }
            return e1[1];
        }
    }
}

File: src/leetcode_grind/day_589.rs
// https://leetcode.com/problems/find-center-of-star-graph/description/?envType=daily-question&envId=2024-06-27
pub fn find_center(edges: Vec<Vec<i32>>) -> i32 {
    use std::collections::HashMap;
    let mut degree = HashMap::new();
    for e in &edges {
        *degree.entry(e[0]).or_insert(0) += 1;
        *degree.entry(e[1]).or_insert(0) += 1;
    }
    for (node, v) in degree {
        if v == edges.len() {
            return node;
        }
    }
    -1
}
